packages/gatsby-source-wordpress/src/gatsby-browser.ts

```diff
@@ -34,6 +33,23 @@ export function onRouteUpdate(): void {
   }
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let ReactDOM: any
+let hasWarnedReact17 = false
+
+try {
+  ReactDOM = require(`react-dom/client`)
+} catch (e) {
+  if (process.env.NODE_ENV === `development` && !hasWarnedReact17) {
+    hasWarnedReact17 = true
+    console.warn(
+      `Upgrade to React 18+ to fix the "Module not found: Can't resolve 'react-dom/client'" warning.`
```

Comment:
Is the React 18 update required? This warning makes it seem like it might be a requirement

`React 18 not detected, falling back to older import method`

Reply:
This is because there's a separate "Module not found:" error in the terminal output that I don't think we can prevent. Even though I'm catching the failed require here Gatsby logs it to stdout. Users on React 17 will see that message. If that terminal warning wasn't there I would let this fail silently

Reply:
Oh also as to the root cause: there were multiple problems. React 18 seems like it was never working - or perhaps it worked in the alpha releases of 18 (that code was added before 18 was released). React 17 seems like it broke at some point because of the `image.parentNode.parentNode` thing. For some reason that caused it to only hydrate images when doing client side navigation and starting on a different page. I'm not entirely sure why that would be but my guess is something about the difference between html hydration vs client rendering